code/controllers/subsystem/tgui.dm

```diff
@@ -27,3 +32,4 @@ var/datum/subsystem/tgui/SStgui
 			ui.process()
 			continue
 		processing_uis.Remove(ui)
+		CHECK_TICK
```

Comment:
This will make it sleep, fucking shit up, i'll pr it to your branch the proper way to do this